dashboard/api_test.py

```diff
@@ -0,0 +1,402 @@
+                course_run=self.course_run
+            )
+        ):
+            api.get_info_for_course(self.user, self.course, None, None)
+        mock_format.assert_called_once_with(self.course_run, api.CourseStatus.OFFERED, self.course)
+
+    @patch('dashboard.api.format_course_for_dashboard', autospec=True)
+    def test_info_not_enrolled_not_passed_not_offered(self, mock_format):
+        """test for get_info_for_course"""
+        with patch(
+            'dashboard.api.get_status_for_courserun',
+            autospec=True,
+            return_value=api.CourseRunUserStatus(
+                status=api.CourseRunStatus.NOT_PASSED,
+                course_run=self.course_run
+            )
+        ), patch('courses.models.Course.get_next_run', autospec=True, return_value=None):
+            api.get_info_for_course(self.user, self.course, None, None)
+        mock_format.assert_called_once_with(None, api.CourseStatus.NOT_OFFERED, self.course)
+
+    @patch('dashboard.api.format_course_for_dashboard', autospec=True)
+    def test_info_grade(self, mock_format):
+        """test for get_info_for_course"""
+        with patch(
+            'dashboard.api.get_status_for_courserun',
+            autospec=True,
+            return_value=api.CourseRunUserStatus(
+                status=api.CourseRunStatus.GRADE,
+                course_run=self.course_run
+            )
+        ):
+            api.get_info_for_course(self.user, self.course, None, None)
+        mock_format.assert_called_once_with(self.course_run, api.CourseStatus.CURRENT_GRADE, self.course)
+
+    @patch('dashboard.api.format_course_for_dashboard', autospec=True)
+    def test_info_read_cert(self, mock_format):
+        """
+        test for get_info_for_course
+        TODO: this particular test need to change when the certificates
+            will be implemented
+        """
+        with patch(
+            'dashboard.api.get_status_for_courserun',
+            autospec=True,
+            return_value=api.CourseRunUserStatus(
+                status=api.CourseRunStatus.READ_CERT,
+                course_run=self.course_run
+            )
+        ):
+            api.get_info_for_course(self.user, self.course, None, None)
+        mock_format.assert_called_once_with(self.course_run, api.CourseStatus.OFFERED, self.course)
+
+    @patch('dashboard.api.format_course_for_dashboard', autospec=True)
+    def test_info_will_attend(self, mock_format):
+        """test for get_info_for_course"""
+        with patch(
+            'dashboard.api.get_status_for_courserun',
+            autospec=True,
+            return_value=api.CourseRunUserStatus(
+                status=api.CourseRunStatus.WILL_ATTEND,
+                course_run=self.course_run
+            )
+        ):
+            api.get_info_for_course(self.user, self.course, None, None)
+        mock_format.assert_called_once_with(self.course_run, api.CourseStatus.CURRENT_GRADE, self.course)
```

Comment:
no clue: I guess it will depend on the implementation of the API that will be used to pull this data. I suspect it will be `None` or `0`